catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Segment.js

```diff
@@ -0,0 +1,464 @@
+
+                var fieldList = [];
+                var prop;
+
+                for (prop in descriptors[segType]) {
+                    if (descriptors[segType].hasOwnProperty(prop)) {
+                        var obj = descriptors[segType][prop];
+                        if (obj.isGroup) {
+                            seg = new Segment.Segment({
+                                segmentName: obj.displayName,
+                                multiValued: obj.multiValued,
+                                parentId: model.get('segmentId'),
+                                segmentType: prop,
+                                containerOnly: FieldDescriptors.isContainerOnly(prop),
+                                nestedLevel: model.get('nestedLevel') + 1,
+                                associationModel: model.get('associationModel'),
+                                autoPopulateFunction: obj.autoPopulateFunction,
+                                autoPopulateId: obj.autoPopulateId,
+                                autoPopulateName: obj.autoPopulateName
+                            });
+                            seg.constructTitle = obj.constructTitle;
+                            var passedModel = dataModel[prop];
+                            if (!passedModel) {
+                                passedModel = seg.get('containerOnly') || obj.multiValued ? [] : {};
+                                dataModel[prop] = passedModel;
+                            }
+                            seg.populateFromModel(passedModel, descriptors);
+                            segs.push(seg);
+                        } else if (!obj.isSlot) {
+                            var field =new Field.FormField({
+                                key: prop,
+                                name: obj.displayName,
+                                desc: obj.description,
+                                type: obj.type,
+                                isSlot: false,
+                                multiValued: obj.multiValued,
+                                value: dataModel[prop] ? dataModel[prop] : obj.values,
+                                min: obj.min,
+                                max: obj.max,
+                                regex: obj.regex,
+                                regexMessage: obj.regexMessage,
+                                possibleValues: obj.possibleValues,
+                                editable: obj.editable,
+                                parentId: model.get('segmentId')
+                            });
+                            field.on('change', field.validate, field);
+                            fieldList.push(field);
+
+                        }
+                    }
+                }
+                addSlotFields(fieldList, dataModel, descriptors[segType], model.get('segmentId'));
+
+                model.set('segments', segs);
+                model.set('fields', fieldList);
```

Comment:
Every time you call set, you trigger a change event and subsequently a render, if something is hooked up to this model. You should be batching all of the model.set calls together and only calling set once with all of the changes.